### PR TITLE
EI-847 Separate out the XA connection factory

### PR DESCRIPTION
**Pull request by madhawa-gunasekara**
Resolves : https://github.com/wso2/product-ei/issues/847

### CODE REVIEW

File: modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQXAConnectionFactory.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+     *
+     * @param username the caller's user name
+     * @param password the caller's password
+     * @return A newly created XAConnection.
+     * @throws JMSException         If creating the XAConnection fails due to some internal error.
+     * @throws JMSSecurityException If client authentication fails due to an invalid user name or password.
+     */
+    public XAConnection createXAConnection(String username, String password) throws JMSException
+    {
+        if (_connectionDetails != null)
```
Comment (milindaperera) [inline modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQXAConnectionFactory.java]: let's remove '_' from _connectionDetails variable name

File: modules/andes-core/client/src/main/java/org/wso2/andes/jndi/PropertiesFileInitialContextFactory.java [hunk 2]
```diff
@@ -307,6 +315,21 @@ protected ConnectionFactory createFactory(String url) throws ConfigurationExcept
     }
 
     /**
+     * Factory method to create new XA Connection Factory instances
+     */
+    protected ConnectionFactory createXAFactory(String url) throws ConfigurationException {
+        try {
+            return new AMQXAConnectionFactory(url);
+        } catch (URLSyntaxException urlse) {
+            _logger.warn("Unable to create xafactory:" + urlse);
```
Comment (milindaperera) [inline modules/andes-core/client/src/main/java/org/wso2/andes/jndi/PropertiesFileInitialContextFactory.java]: let's rename "_logger" removing  underscore

File: modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQXAConnectionFactory.java (hunk 1 above)
Comment (madhawa-gunasekara) [reply modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQXAConnectionFactory.java]: @milindaperera: this is a forked code, therefore IMO It's not good to modify the code.

Comment (milindaperera) [reply modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQXAConnectionFactory.java]: +1

**Pull request merged**